frame/nomination-pools/src/lib.rs

```diff
@@ -1827,6 +1836,50 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Update the roles of the pool.
+		///
+		/// The root is the only entity that can change any of the roles, including itself,
+		/// excluding the depositor, who can never change.
+		///
+		/// It emits an event, notifying UIs of the role change. This event is quite relevant to
+		/// most pool members and they should be informed of changes to pool roles.
+		#[pallet::weight(T::WeightInfo::update_roles())]
+		pub fn update_roles(
+			origin: OriginFor<T>,
+			pool_id: PoolId,
+			root: ConfigOp<T::AccountId>,
+			nominator: ConfigOp<T::AccountId>,
+			state_toggler: ConfigOp<T::AccountId>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let mut bonded_pool = BondedPool::<T>::get(pool_id).ok_or(Error::<T>::PoolNotFound)?;
+			ensure!(bonded_pool.roles.root == who, Error::<T>::DoesNotHavePermission);
```

Comment:
if you can forsee it being useful, could also make this callable by `ROOT` origin too.

Reply:
b70a514f8e94b74af4475d8f03fb53c7b8affe3a